File: src/main/java/com/indiepost/repository/BookmarkRepository.java
package com.indiepost.repository;

import com.indiepost.model.Bookmark;

import java.util.List;

public interface BookmarkRepository {
    void save(Bookmark bookmark);

    void create(Long userId, Long postId);

    void delete(Bookmark bookmark);

    Bookmark findOneByUserIdAndPostId(Long userId, Long postId);

    List<Bookmark> findByUserIdAndPostIds(Long userId, List<Long> postIds);

    void removeAllBookmarksByUserId(Long userId);
}

File: src/main/kotlin/com/indiepost/repository/BookmarkRepository.kt
package com.indiepost.repository

import com.indiepost.model.Bookmark

interface BookmarkRepository {
    fun save(bookmark: Bookmark)

    fun create(userId: Long, postId: Long)

    fun delete(bookmark: Bookmark)

    fun findOneByUserIdAndPostId(userId: Long, postId: Long): Bookmark?

    fun findByUserIdAndPostIds(userId: Long, postIds: List<Long>): List<Bookmark>

    fun removeAllBookmarksByUserId(userId: Long)
}
